docs/source/client.rst

```diff
@@ -112,8 +112,14 @@ For more information see the page on :doc:`Managing Computation <manage-computat
 Pure Functions by Default
 -------------------------
 
-By default we assume that all functions are pure_.  If this is not the case we
-should use the ``pure=False`` keyword argument.
+By default, ``distributed`` assumes that all functions are pure_.
+
+Pure functions:
+
+* always return the same output for a given set of inputs
+* do not have side effects, like modifying global state or creating files
+
+If this is not the case, you should use the ``pure=False`` keyword argument.
```

Comment:
I realize this predates this PR, but could you mention where users can specify the `pure=` keyword (i.e. `client.submit` and `client.map`)

Reply:
added in https://github.com/dask/distributed/pull/4644/commits/7a207260b9f31efd0c0a5d7d36ccbf836a519135